backend/lucene/src/main/java/org/hibernate/search/backend/lucene/lowlevel/docvalues/impl/DoubleMultiValuesToSingleValuesSource.java

```diff
@@ -155,11 +197,17 @@ protected NumericDoubleValues select(SortedNumericDoubleDocValues values, final
 		return new NumericDoubleValues() {
 
 			int lastSeenParentDoc = -1;
-			double lastEmittedValue = -1;
+			private Iterator<Double> value;
+			private List<Double> all;
 
 			@Override
 			public double doubleValue() {
-				return lastEmittedValue;
+				return value != null ? value.next() : -1;
```

Comment:
NumericDoubleValues are not supposed to be multi-valued, on contrary to SortedNumericDocValues.
This hijacks and interface that was not intended for this purpose at all.

Worse, in aggregations with a multi-value mode of none, this solution takes a `SortedNumericDocValues` (multi-valued), converts it to a `NumericDoubleValues` (single-valued) thanks to `DoublieMultiValuesToSingleValuesSource`, and then uses this hack to actually extract multiple values and expose them as multi-valued in `LongMultiDoubleValuesSource`...

This is clearly sub-optimal, and a (way) better solution would be to simply use `SortedNumericDocValues` directly in `LongMultiDoubleValuesSource`.

I will work on removing this hack before I submit a PR for aggregations on multi-valued fields.

Reply:
It is not necessary for sorting, it always assumes cout = 1. But for aggregation, when it should select all non-redundant values. This simple method eliminates code redundancy. Earlier I had a different solution, but ultimately it turned out to be optimal. Look at the aggregation problem when multivalue is none. Currently, Lucene does not solve this problem. multivalue is new at all. I suspect that over time they will solve, so far a simple modification allows aggregation of many values. This is not a great momentary modification. So far, it makes us ahead of lucene, with time lucene will probably find the optimal solution and it will be possible in higher versions of lucene to go back and use the original solution.

Reply:
Also, I was looking for a different solution, but as of today, it required the least effort, copying the overall implementation for aggregation and adding loops for many values. Currently, thanks to this I am working on a slightly modified package with the "guy" lucene. Maybe in the future lucene will solve the problem, so far not much is happening in 8.5. And the patch bypasses the problem and makes aggregation work properly. Removing the patch will prevent aggregation of many values without converting it to SUM, AVG, MIN, MAX as NONE aggregation.

Reply:
One could try to modify "FacetsCollector" for non-converted multiple values accordingly. Then it could have been done without doing and modifying the classes from the lucene "fecets" package. But it would be rather difficult to do it, maybe easier for nested values, but for flat relations it would be a big problem.